src/main/java/com/smartbear/tcpbench/engines/Comet.java

```diff
@@ -51,13 +48,20 @@ public void prepare(String projectName) throws ApiException {
     @Override
     public void defineTestCycle(String testCycleId, List<Verdict> verdicts, Query query) {
         List<String> shas = query.getShas(testCycleId);
-        Set<String> patches = shas.stream().flatMap(sha -> query.getModifiedFiles(sha, ".java$").stream()).collect(Collectors.toSet());
-        int filesChanged = patches.size();
-
+        Changes changes = new Changes();
+        if (shas.size() > 0) {
```

Comment:
I'm not sure about this one. `query.getShas(testCycleId)` only returns a single sha for most test cycles. That is, the git sha the Travis build ran for.

I discovered that the RTPTorrent dataset sometimes maps multiple shas to the same travis job id. It only happens a few times, and it seems like an anomaly. Most test cycles only have a single sha.

So in order to get the changes, I think it would be better to use the parent of the single sha. Otherwise you're going to end up with empty changes most of the time.

Reply:
I see, I made this choice because I interpret multiple shas as multiple commit between to travis job and I wanted to manage that, but maybe is anomaly as you said?

Empty changes indicate that there is no change in the code between two cycles (same commit), I don't think it's an issue to have empty values if it represent the reality.

However it's indeed a problem for `sinceLast` which should never be empty because jobs are never executed at the same time.